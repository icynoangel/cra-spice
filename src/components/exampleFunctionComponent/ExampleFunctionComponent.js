import './ExampleFunctionComponent.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';

export const MULTIPLY_FACTOR = 10;

const ExampleFunctionComponent = ({ value, intl }) => {
  const multipliedValue = value * MULTIPLY_FACTOR;
  return (
    <div role="article" className="example-function-component">
      {intl.formatMessage({
        id: 'multipliedValue',
        defaultMessage: 'Multiplied Value'
      })}{' '}
      {multipliedValue}
    </div>
  );
};

ExampleFunctionComponent.propTypes = {
  value: PropTypes.number,
  intl: PropTypes.object
};

export default injectIntl(ExampleFunctionComponent);
