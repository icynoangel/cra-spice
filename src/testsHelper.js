import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

import dictionaryMock from './api/mocks/dictionary';

const intlProvider = new IntlProvider(
  { locale: dictionaryMock.locale, messages: dictionaryMock.messages },
  {}
);
const { intl } = intlProvider.getChildContext();

export default intl;

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node, { context, ...additionalOptions } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    ...additionalOptions
  });
}

export function mountWithIntl(
  node,
  { context, childContextTypes, ...additionalOptions } = {}
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign(
      {},
      { intl: intlShape },
      childContextTypes
    ),
    ...additionalOptions
  });
}
