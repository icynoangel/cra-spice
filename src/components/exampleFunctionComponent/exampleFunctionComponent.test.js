import React from 'react';
import ExampleFunctionComponent from './exampleFunctionComponent';
import { shallowWithIntl } from '../../testsHelper';
import enzymeToJson from 'enzyme-to-json';

describe('<ExampleFunctionComponent />', function () {
  beforeEach(() => {
    this.value = 10;

    this.getComponent = () => {
      return <ExampleFunctionComponent.WrappedComponent value={this.value} />;
    };
  });

  it('should render <ExampleFunctionComponent />', () => {
    const wrapper = shallowWithIntl(this.getComponent());
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});
