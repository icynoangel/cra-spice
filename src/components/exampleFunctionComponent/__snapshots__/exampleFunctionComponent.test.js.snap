// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleFunctionComponent /> should render <ExampleFunctionComponent /> 1`] = `
<div
  className="example-function-component"
>
  Multiplied Value
   
  100
</div>
`;
